plugins/battery/battery.plugin.sh

```diff
@@ -6,7 +6,7 @@ if ! command -v "command_exists" >/dev/null; then	command_exists() { command -v
 ac_adapter_connected(){
   if command_exists upower;
   then
-    upower -i $(upower -e | grep BAT) | grep 'state' | grep -q 'charging\|fully-charged'
+    upower -i $(upower -e | grep BAT) | grep 'state' | grep -q '[^(dis)]charging\|fully-charged'
```

Comment:
`[^(dis)]` doesn't work as you expect. It requires one character that is not one of `(`, `d`, `i`, `s`, or `)` before the word `charging`.

By the way, this is not related to this PR, but `\|` is a non-POSIX extension to BRE. We can use newlines to separate multiple targets of `grep`. Or, we can specify the `-E` option to `grep` and use ERE.

Reply:
Alternatively, couldn't `-e` be used?

You could do
```bash
grep -q -e '^(dis)?charging' -e 'fully-charged'
```
which would accomplish the same thing

Reply:
> Alternatively, couldn't `-e` be used?

Ah, yes, you are right. That is more common.

> ```
> grep -q -e '^(dis)?charging' -e 'fully-charged'
> ```

Unless you specify the `-E` option, `(dis)?` matches with the literal five characters "`(dis)?`", which I guess is not your intent. If you specify `-E`, you could have simply used `|`. If you don't specify `-E`, you need to write it in BRE, i.e., `grep -q -e '^\(dis\)\{0,1\}charging' -e 'fully-charged'`.

But actually, what we want to match here is not "`discharging` or `charging` in the beginning of line" but "`charging`, which is not a part of `discharging`, in any position in the line". The simplest solution might be adding `grep -v`, but entirely rewriting it in Awk might be better.